packages/esm-patient-chart-app/src/ui-components/action-menu.component.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useCallback } from 'react';
+import { ExtensionSlot, useLayoutType } from '@openmrs/esm-framework';
+import Button from 'carbon-components-react/es/components/Button';
+import { HeaderPanel } from 'carbon-components-react';
+
+import { Edit20 } from '@carbon/icons-react';
+import styles from './action-menu.scss';
+import { isDesktop } from '../utils';
+
+interface ActionMenuInterface {
+  open: boolean;
+}
+
+export const CHARTS_DRAWER_SLOT = 'drawer-slot';
+export const CHARTS_ACTION_MENU_ITEMS_SLOT = 'action-menu-items-slot';
+
+export const ActionMenu: React.FC<ActionMenuInterface> = ({ open }) => {
+  const layout = useLayoutType();
+
+  const menu = isDesktop(layout) ? (
+    <aside className={styles.rightSideNav}>
+      <ExtensionSlot extensionSlotName={CHARTS_ACTION_MENU_ITEMS_SLOT} />
+    </aside>
+  ) : (
+    <button className={styles.actionBtn}>
+      <Edit20 />
+    </button>
+  );
+
+  return (
+    <>
+      {menu}
+      <HeaderPanel className={styles.actionPanel} expanded={open}>
+        <ExtensionSlot extensionSlotName={CHARTS_DRAWER_SLOT} />
```

Comment:
so i'm creating two extension slots; one for the menu items and one for the drawer content. Figuring out how to include all the state for it would probably require to usage of context api or something else... regardless, here's just the foundations

Reply:
Hm what states are you referring to?

Reply:
You can pass on `props` via the `state` prop.

Reply:
yea, exactly - where that state will be is just gonna be up to the user. i didn't bother baking in that state logic - just added the open prop on the action menu